overlord/hookstate/ctlcmd/system_mode.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/strutil"
+	"gopkg.in/yaml.v2"
+)
+
+type systemModeCommand struct {
+	baseCommand
+}
+
+var shortSystemModeHelp = i18n.G("Get the current system mode information")
+
+var longSystemModeHelp = i18n.G(` The system-mode command returns
+ information about system mode the device is in, including the mode
+ itself, whether the model snaps have been seeded, and whether the
+ device is in factory mode. Its output is in YAML format.  The system
+ mode will be one of run, recover, or install.
+ Example output:
+
+ $ snapctl system-mode
+ system-mode: install
+ seeded: true
+ factory: true
+`)
+
+func init() {
+	addCommand("system-mode", shortSystemModeHelp, longSystemModeHelp, func() command { return &systemModeCommand{} })
+}
+
+var devicestateSystemModeInfoFromState = devicestate.SystemModeInfoFromState
+
+type systemModeResult struct {
+	SystemMode string `yaml:"system-mode,omitempty"`
+	Seeded     bool   `yaml:"seeded"`
+	Factory    bool   `yaml:"factory,omitempty"`
```

Comment:
No `host-ubuntu-data` in the output ?

Reply:
yes, I think it's more useful in "snap system-mode" than here given that hooks can't really usually write there, we might add it for symmetry but when we have it in the other place. So I tried to be cautious for now.